keadm/cmd/keadm/app/cmd/util/common.go

```diff
@@ -675,3 +650,46 @@ func askForconfirm() (bool, error) {
 		return false, fmt.Errorf("Invalid Input")
 	}
 }
+
+func downloadServiceFile(componentType types.ComponentType, version semver.Version, storeDir string) error {
+	//No need to download if
+	//1. the systemd not exists
+	//2. the version is less than 1.1.0
+	//3. the service file already exists
+	if hasSystemd() && version.GE(semver.MustParse("1.1.0")) {
+		var ServiceFileName string
+		switch componentType {
+		case types.CloudCore:
+			ServiceFileName = CloudServiceFile
+			if version.EQ(semver.MustParse("1.1.0")) {
+				fmt.Println("[Run as service]skip download service file for cloudcore-v1.1.0, not support")
+				return nil
+			}
+		case types.EdgeCore:
+			ServiceFileName = EdgeServiceFile
+			if version.EQ(semver.MustParse("1.1.0")) {
+				ServiceFileName = OldEdgeServiceFile
+			}
+		default:
+			return fmt.Errorf("component type %s not support", componentType)
+		}
+		ServiceFilePath := storeDir + "/" + ServiceFileName
+		strippedVersion := fmt.Sprintf("%d.%d", version.Major, version.Minor)
```

Comment:
remove tmp val

Reply:
sorry, what does "temp val" mean here?

Reply:
strippedVersion.

Reply:
But I still think this temp variable is needed, or line about constructing serviceFileURL would be very long. 
: )